packages/sdks/snippets/react/src/routes/blueprints/ProductEditorial.tsx

```diff
@@ -0,0 +1,48 @@
+import {
+  BuilderContent,
+  Content,
+  fetchOneEntry,
+  isPreviewing,
+} from '@builder.io/sdk-react';
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import ProductFooter from '../../components/ProductFooter';
+import ProductHeader from '../../components/ProductHeader';
+import ProductInfo from '../../components/ProductInfo';
+
+const API_KEY = 'ee9f13b4981e489a9a1209887695ef2b';
+const MODEL_NAME = 'product-editorial';
+
+export default function ProductEditorial() {
+  const [product, setProduct] = useState(null);
+  const [editorial, setEditorial] = useState<BuilderContent | null>(null);
+  const { id } = useParams();
+
+  useEffect(() => {
+    Promise.all([
+      fetch(`https://fakestoreapi.com/products/${id}`).then((res) =>
+        res.json()
+      ),
+      fetchOneEntry({
+        model: MODEL_NAME,
+        apiKey: API_KEY,
+        userAttributes: { urlPath: window.location.pathname },
+      }),
+    ]).then(([productData, editorialData]) => {
+      setProduct(productData);
+      setEditorial(editorialData);
+    });
+  }, [id]);
```

Comment:
- `Promise.all` is an unnecessary complication. There is nothing gained from running the promises together 
- the hook is dependent on `id` but only one of the calls is tied to the `id`

should be two separate hooks:

```suggestion
  useEffect(() => {
    fetch(`https://fakestoreapi.com/products/${id}`)
      .then((res) => res.json())
      .then((productData) => {
        setProduct(productData);
      });
  }, [id]);

  useEffect(() => {
    fetchOneEntry({
      model: MODEL_NAME,
      apiKey: API_KEY,
      userAttributes: { urlPath: window.location.pathname },
    }).then((editorialData) => {
      setEditorial(editorialData);
    });
  }, []);
```